### PR TITLE
Update Insteon component configuration

### PR DESCRIPTION
**Pull request by klaasnicolaas**
**Description:**
Update style of Insteon component documentation to follow new configuration variables description.
Related to #6385.

**Note:**
Check carefully that everything is correct, it was a hell or a job to put everything right.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [ ] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/insteon.markdown [hunk 1]
```diff
@@ -81,129 +75,136 @@ insteon:
...
+  description: The username to login in to the local Hub. You can find your Hub username on the bottom of the Hub or you can use the Insteon app.
+  required: Required for Hub version 2 setup
+  type: string
+password:
+  description: The password to login in to the local Hub. You can find your Hub password on the bottom of the Hub or you can use the Insteon app.
+  required: Required for Hub version 2 setup
+  type: string
+hub_version:
+  description: The Hub version number where model [2242] is Hub version 1 and model [2245] is Hub version 2.
+  required: Required for Hub version 1 setup
```
Comment (frenck) [inline source/_components/insteon.markdown:102]: `required` should be a `true`/`false`/`inclusive`/`exclusive`.

File: source/_components/insteon.markdown [hunk 2]
```diff
@@ -81,129 +75,136 @@ insteon:
...
+  description: The host name or IP address of the Hub.
+  required: Required for Hub setup
+  type: string
+ip_port:
+  description: The IP port number of the Hub. For Hub model [2245] (i.e. Hub version 2) the default port is 25105. For the Hub model [2242] (i.e. Hub version 1) the default port is 9761. Use the Insteon app to find the port number for your specific Hub.
+  required: Optional for Hub setup
+  type: integer
+username:
+  description: The username to login in to the local Hub. You can find your Hub username on the bottom of the Hub or you can use the Insteon app.
+  required: Required for Hub version 2 setup
```
Comment (frenck) [inline source/_components/insteon.markdown:94]: `required` should be a `true`/`false`/`inclusive`/`exclusive`.

File: source/_components/insteon.markdown [hunk 3]
```diff
@@ -81,129 +75,136 @@ insteon:
...
+  type: string
+password:
+  description: The password to login in to the local Hub. You can find your Hub password on the bottom of the Hub or you can use the Insteon app.
+  required: Required for Hub version 2 setup
```
Comment (frenck) [inline source/_components/insteon.markdown:98]: `required` should be a `true`/`false`/`inclusive`/`exclusive`.

File: source/_components/insteon.markdown [hunk 4]
```diff
@@ -81,129 +75,136 @@ insteon:
...
-  to the X10 standard command for All Units Off.
-- **x10_all_lights_on** (*Optional*):  Creates a binary_sensor that responds
-  to the X10 standard command for All Lights On
-- **x10_all_lights_off** (*Optional*): Creates a binary_sensor that responds
-  to the X10 standard command for All Lights Off
+
+{% configuration %}
+port:
+  description: The serial or USB port for your device, e.g., `/dev/ttyUSB0` or `COM3`.
+  required: Required for PLM setup
```
Comment (frenck) [inline source/_components/insteon.markdown:82]: `required` should be a `true`/`false`/`inclusive`/`exclusive`.

File: source/_components/insteon.markdown [hunk 5]
```diff
@@ -81,129 +75,136 @@ insteon:
...
+  type: string
+host:
+  description: The host name or IP address of the Hub.
+  required: Required for Hub setup
```
Comment (frenck) [inline source/_components/insteon.markdown:86]: `required` should be a `true`/`false`/`inclusive`/`exclusive`.

**Review by frenck — CHANGES_REQUESTED** [review]
This must have been a pain to rewrite/fix!
Sorry for these comments I've left...

File: source/_components/insteon.markdown (hunk 1 above)
Comment (klaasnicolaas) [reply source/_components/insteon.markdown:102]: why does it say in the documentation that you can put a string? [Info](https://developers.home-assistant.io/docs/en/documentation_create_page.html)

Comment (frenck) [reply source/_components/insteon.markdown:102]: So why am I the one that didn't know? 😭 

Sorry about that m8!

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks, @klaasnicolaas!

Sorry about the misjudgment on my end.
Great job rewriting this! 👍 

Happy Hacktoberfest 🎉

File: source/_components/insteon.markdown (hunk 1 above)
Comment (klaasnicolaas) [reply source/_components/insteon.markdown:102]: Well everyone learns something at any time 😉

**Pull request merged**